### PR TITLE
fix: support multi-line KV for secrets, envvars, and labels

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes https://github.com/google-github-actions/deploy-cloud-functions/issues/308

### CODE REVIEW

**Comment by verbanicm** [comment]
@sethvargo Somethings wrong here....

File: src/client.ts [hunk 1]
```diff
@@ -509,12 +498,19 @@ export class CloudFunctionsClient {
...
 
     try {
-      await CloudFunctionsClient.request('PUT', uploadURL, zipFile, {
-        headers: {
-          'content-type': 'application/zip',
-          'x-goog-content-length-range': '0,104857600',
-        },
+      // This is different logic than the primary request function, and it does
+      // not return JSON.
+      const client = new HttpClient(userAgent);
```
Comment (verbanicm) [inline src/client.ts:503]: This can make use of `this.#client.request` rather than creating a new client

**Review by verbanicm — APPROVED** [review]

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -265,7 +265,9 @@ jobs:
         env_vars_file: './tests/env-var-files/test.good.yaml'
         build_environment_variables: 'FOO=bar, ZIP=zap'
         build_environment_variables_file: './tests/env-var-files/test.good.yaml'
-        secret_environment_variables: 'FOO=${{ secrets.DEPLOY_CF_SECRET_VERSION_REF }},BAR=${{ secrets.DEPLOY_CF_SECRET_REF }}'
+        secret_environment_variables: |-
+          FOO=${{ secrets.DEPLOY_CF_SECRET_VERSION_REF }}
+          BAR=${{ secrets.DEPLOY_CF_SECRET_REF }}'
```
Comment (bharathkkb) [inline .github/workflows/integration.yml:270]:
```suggestion
          BAR=${{ secrets.DEPLOY_CF_SECRET_REF }}
```

Comment (bharathkkb) [reply .github/workflows/integration.yml:270]: I think this is causing the test failure

**Pull request merged**